open-notes/maturity-model.md

```diff
@@ -0,0 +1,394 @@
+---
+id: '202107270936'
+tags: []
+related: []
+from:
+---
+
+# Maturity Model
+
+This is something that each org or team could come up by themselves depending on the their objectives. My feeling is that in an organisation having these standard across teams of the same type ie. dev, infra, etc. could be beneficial for standardizing in a way that still fosters autonomy.
+
+This goes toward something I believe quite strongly. Standardizing when you have immature standards is a good way to get stuck in mediocrity, at best. Unfortunately, you cannot know what you do not know, so my hope is that publishing and crowd-sourcing these maturity models across organisations will lead to better outcomes.
+
+A maturity model is made up of levels. Each of these levels have various skills in them. I find skill a good word for these as usually anything that brings benefit to a team requires practice. So the skill is not just about doing something, but practicing it until you are skillful at it.
+Take unit testing as an example. You can have some unit tests in your project and say you have that Skill. Having unit tests in your project is different from having maintainable unit tests that test useful aspects of your codebase. If they are written mostly after the fact and submitted as completely different PRs, they are not really helping to catch bugs. This is a sign of chasing a test coverage metric rather than using unit testing as a skill to deliver higher quality software.
+
+And here in lies the danger in having a checklist like this maturity model. It can be viewed as a todo list to check-off, maybe even a source of pride over other teams. This is not the intent! It is meant as a guide to becoming a more skillful team at delivering quality software that is a joy to work with and adds value to customers.
+
+
+## Tags
+
+I am proposing some tags that can be applied to each Skill. This allows us to chose a mix of Skills that stretch the team in different ways at each level.
+
+Examples:  
+
+- code
+- knowledge
+- process
+- observability
+- maintenance
+- security
+- communication
+
+## Example
+
+### General development team
```

Comment:
I feel that it is focused on engineering. I'm missing product and architecture, the other 2 tenents of a modern product engineering team, that allows a digital organisation to innovate.